compiler/rustc_ast/src/ast_traits.rs

```diff
@@ -313,6 +314,11 @@ impl<Wrapped, Tag> AstNodeWrapper<Wrapped, Tag> {
     pub fn new(wrapped: Wrapped, _tag: Tag) -> AstNodeWrapper<Wrapped, Tag> {
         AstNodeWrapper { wrapped, tag: Default::default() }
     }
+
+    pub fn from_mut(wrapped: &mut Wrapped, _tag: Tag) -> &mut AstNodeWrapper<Wrapped, Tag> {
+        // SAFETY: `AstNodeWrapper` is `repr(transparent)` w.r.t `Wrapped`
+        unsafe { <*mut Wrapped>::from(wrapped).cast::<Self>().as_mut().unwrap() }
```

Comment:
```suggestion
        unsafe { transmute(wrapped) }
```
Is this the idiomatic way to do this?
A transmute would be *much* more clear and obvious (for me, at least).

Reply:
I've played with these APIs a bit, I think
```suggestion
        unsafe { &mut *<*mut Wrapped>::cast(s) }
```
would be the best looking version (`cast` instead of `transmute`, but still no conversion method noise).